javaslang-benchmark/src/test/java/javaslang/benchmark/collection/TreeMapBenchmark.java

```diff
@@ -0,0 +1,398 @@
+        Stack<Integer> values = List.empty();
+        for (Map.Entry<Integer, String> aTreeMap : state.javaTreeMap.entrySet()) {
+            values.push(aTreeMap.getKey());
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("IterateAscending")
+    public Object iterateAscending_Scala(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        Iterator<scala.Tuple2<Integer, String>> iterator = state.scalaTreeMap.iterator();
+        while (iterator.hasNext()) {
+            values.push(iterator.next()._1);
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("IterateDescending")
+    public Object iterateDescending_Slang(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        for (Tuple2<Integer, String> aTreeMap : state.slangTreeMap.descendingIterator()) {
+            values.push(aTreeMap._1);
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("IterateDescending")
+    public Object iterateDescending_Java(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        for (Map.Entry<Integer, String> aTreeMap : state.javaTreeMap.descendingMap().entrySet()) {
+            values.push(aTreeMap.getKey());
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("SubMap")
+    public Object createSubMap_WithOneThirdElements_AndIterate_Slang(BenchmarkState state) {
+        // We will create subset with 1/3 of entries in the original map
+        int keyOffset = state.numItems / 6;
+        SortedMap<Integer, String> subMap = state.slangTreeMap.subMap(state.keysSorted.get(keyOffset), true, state.keysSorted.get(state.numItems - keyOffset), true);
+        List<Integer> values = List.empty();
+        for (Tuple2<Integer, String> entry : subMap) {
+            values.push(entry._1);
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("SubMap")
+    public Object createSubMap_WithOneThirdElements_AndIterate_Java(BenchmarkState state) {
+        // We will create subset with 1/3 of entries in the original map
+        int keyOffset = state.numItems / 6;
+        NavigableMap<Integer, String> subMap = state.javaTreeMap.subMap(state.keysSorted.get(keyOffset), true, state.keysSorted.get(state.numItems - keyOffset), true);
+        List<Integer> values = List.empty();
+        for (Map.Entry<Integer, String> entry : subMap.entrySet()) {
+            values.push(entry.getKey());
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("SubMap")
+    public Object createDescendingSubMap_WithOneThirdElements_AndIterate_Slang(BenchmarkState state) {
+        // We will create subset with 1/3 of entries in the original map
+        int keyOffset = state.numItems / 6;
+        SortedMap<Integer, String> subMap = state.slangTreeMap.descendingMap().subMap(state.keysSorted.get(state.numItems - keyOffset), true, state.keysSorted.get(keyOffset), true);
+        assert subMap.size() > 0;
```

Comment:
hmm, why do we have an assert here, but not elsewhere?
Also, there may be asserts in the production code also, it's the reason I disabled assertions and changed these to checks and exceptions.